pkg/crypto/aes/aes.go

```diff
@@ -0,0 +1,86 @@
+package aes
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/hex"
+	"bytes"
+	"github.com/Shopify/go-lua"
+	"io"
+)
+
+func Open(l *lua.State) {
+	aesOpen := func(l *lua.State) int {
+		lua.NewLibrary(l, aesLibrary)
+		return 1
+	}
+	lua.Require(l, "goluago/crypto/aes", aesOpen, false)
+	l.Pop(1)
+}
+
+var aesLibrary = []lua.RegistryFunction{
+	{"encryptCBC", encryptCBC},
+	{"decryptCBC", decryptCBC},
+}
+
+func encryptCBC(l *lua.State) int {
+	key, _ := hex.DecodeString(lua.CheckString(l, 1))
+	plaintext := []byte(lua.CheckString(l, 2))
+
+	block, err := aes.NewCipher(key)
+	if err != nil {
+		lua.Errorf(l, err.Error())
+		panic("unreachable")
+	}
+
+	content := PKCS5Padding(plaintext, aes.BlockSize, len(plaintext))
+
+	ciphertext := make([]byte, aes.BlockSize + len(content))
+	iv := ciphertext[:aes.BlockSize]
+	if _, err := io.ReadFull(rand.Reader, iv); err != nil {
+		lua.Errorf(l, err.Error())
+		panic("unreachable")
+	}
+
+	encrypter := cipher.NewCBCEncrypter(block, iv)
+	encrypter.CryptBlocks(ciphertext[aes.BlockSize:], content)
+
+	l.PushString(hex.EncodeToString(ciphertext))
+
+	return 1
+}
+
+func decryptCBC(l *lua.State) int {
+	key, _ := hex.DecodeString(lua.CheckString(l, 1))
+	ciphertext, _ := hex.DecodeString(lua.CheckString(l, 2))
+
+	block, err := aes.NewCipher(key)
+	if err != nil {
```

Comment:
This is a great example of how we could report errors in other places where errors are dropped 👌🏻